bundles/org.openhab.binding.sedif/src/main/java/org/openhab/binding/sedif/internal/handler/BridgeSedifWebHandler.java

```diff
@@ -0,0 +1,297 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.openhab.binding.sedif.internal.api.SedifHttpApi;
+import org.openhab.binding.sedif.internal.config.SedifBridgeConfiguration;
+import org.openhab.binding.sedif.internal.constants.SedifBindingConstants;
+import org.openhab.binding.sedif.internal.discovery.SedifDiscoveryService;
+import org.openhab.binding.sedif.internal.dto.AuraContext;
+import org.openhab.binding.sedif.internal.dto.AuraResponse;
+import org.openhab.binding.sedif.internal.dto.Contract;
+import org.openhab.binding.sedif.internal.dto.Contracts;
+import org.openhab.binding.sedif.internal.dto.Event;
+import org.openhab.binding.sedif.internal.helpers.SedifListener;
+import org.openhab.binding.sedif.internal.types.SedifException;
+import org.openhab.core.auth.client.oauth2.OAuthFactory;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.io.net.http.TrustAllTrustManager;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingRegistry;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * {@link BridgeSedifWebHandler} is the base handler to access sedif data.
+ *
+ * @author Laurent Arnal - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class BridgeSedifWebHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(BridgeSedifWebHandler.class);
+    protected @Nullable SedifBridgeConfiguration config;
+
+    private Set<SedifListener> listeners = new CopyOnWriteArraySet<>();
+
+    private Dictionary<String, Contract> contractDict = new Hashtable<String, Contract>();
+
+    protected final HttpService httpService;
+    protected final BundleContext bundleContext;
+    protected final HttpClient httpClient;
+    protected final SedifHttpApi sedifApi;
+    protected final ThingRegistry thingRegistry;
+
+    protected final Gson gson;
+    private String sid = "";
+
+    private @Nullable AuraContext appCtx;
+    private @Nullable String token = "";
+
+    protected boolean connected = false;
+
+    private static final int REQUEST_BUFFER_SIZE = 8000;
+    private static final int RESPONSE_BUFFER_SIZE = 200000;
+
+    public BridgeSedifWebHandler(Bridge bridge, final @Reference HttpClientFactory httpClientFactory,
+            final @Reference OAuthFactory oAuthFactory, final @Reference HttpService httpService,
```

Comment:
remove oauthfactory.

I think this class can use some clean up for this kind of properties and parameters